src/scriptdev2/scripts/eastern_kingdoms/molten_core/boss_ragnaros.cpp

```diff
@@ -169,6 +167,58 @@ struct boss_ragnarosAI : public Scripted_NoMovementAI
             m_uiEnterCombatTimer = 10000;
     }
 
+    // Custom threat management for targets in melee range
+    bool CanMeleeTargetInRange()
```

Comment:
Cant this be substituted by SelectAttackingTarget - ATTACKING_TARGET_TOPAGGRO and SELECT_FLAG_IN_MELEE_RANGE?